### PR TITLE
.github/workflows: add cosmos/gosec vulnerability scanner for each Push/PR

### PR DESCRIPTION
**Pull request by odeke-em**
Adds a code vulnerability scanner that'll flag issues and issue advisories from cosmos/gosec https://github.com/cosmos/gosec

### CODE REVIEW

**Comment by odeke-em** [comment]
Kind FYI for @thanethomson @elias-orijtech @marbar3778 @kirbyquerby @willpoint @cmwaters

**Comment by cmwaters** [comment]
Hey @odeke-em, can you open this PR against `main`, the default branch. Thanks!

**Comment by odeke-em** [comment]
Gotcha, thanks for letting me know @cmwaters, I've made the PR against the main branch. Please take a look again.

File: .github/workflows/gosec.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Run Gosec
+on:
+  pull_request:
+    paths:
+      - '**/*.go'
+      - 'go.mod'
+      - 'go.sum'
+  push:
+    branches:
+      - main
```
Comment (sergio-mena) [inline .github/workflows/gosec.yml:10]: Is it possible to add all branches under `feature`? (i.e., `feature/*` or something similar)

Comment (sergio-mena) [reply .github/workflows/gosec.yml:10]: Also, any reason for not running this on throttle branches?

File: .github/workflows/gosec.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    env:
+      GO111MODULE: on
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v3
+
+      - name: Run Gosec Security Scanner
+        uses: cosmos/gosec@master
+        with:
+          # we let the report trigger content trigger a failure using the GitHub Security features.
```
Comment (sergio-mena) [inline .github/workflows/gosec.yml:31]: Could you reword to avoid repeating "trigger" twice? I actually don't fully understand that part of the sentence

**Review by sergio-mena — APPROVED** [review]
Apart from minor nits, LGTM.
Thanks for this, @odeke-em !

File: .github/workflows/gosec.yml (hunk 1 above)
Comment (thanethomson) [inline .github/workflows/gosec.yml:10]: @odeke-em would it be valuable to run this on feature and release branches? e.g.:

```suggestion
      - main
      - 'feature/*'
      - v0.37.x
      - v0.34.x
```

Comment (odeke-em) [reply .github/workflows/gosec.yml:10]: Sounds good to me! Thank you @thanethomson

Comment (odeke-em) [reply .github/workflows/gosec.yml:10]: @thanethomson made suggestions for the branches so I added them. However, y'all free to add as many branches after we've merged this PR :-)

Comment (odeke-em) [reply .github/workflows/gosec.yml:10]: Thank you, I committed this suggestion @thanethomson.

File: .github/workflows/gosec.yml (hunk 2 above)
Comment (odeke-em) [reply .github/workflows/gosec.yml]: Done, thank you!

**Comment by odeke-em** [comment]
Thank you for the reviews @thanethomson @sergio-mena! I've made updates, please take a look.

**Review by thanethomson — APPROVED** [review]
I'm interested to try this out and see if it works for us.

@odeke-em could you please pull in the latest changes from `main`? I don't think we have permission to do that for your PR. Once all tests pass, Mergify should automatically merge this.

**Comment by odeke-em** [comment]
I've rebased with the latest from main, thank you @thanethomson!

**Pull request merged**